### PR TITLE
Internalize types 2

### PR DESCRIPTION
**Pull request by cptjazz**
This PR removes a bunch of methods/types that were marked `Obsolete` for quite some time and were unused. Some `Obsolete` types that still had internal usages were made `internal`.

Furthermore I cleaned up a bit the types related to `ICancelable`.

Related to #4223

### CODE REVIEW

File: src/core/Akka/Actor/Scheduler/DedicatedThreadScheduler.cs [hunk 1]
```diff
@@ -21,7 +20,7 @@ namespace Akka.Actor
     /// Obsolete. Use <see cref="HashedWheelTimerScheduler"/> instead.
     /// </summary>
     [Obsolete("Replaced with HashedWheelTimerScheduler [1.1.2]")]
-    public class DedicatedThreadScheduler : SchedulerBase, IDateTimeOffsetNowTimeProvider, IDisposable
+    internal class DedicatedThreadScheduler : SchedulerBase, IDateTimeOffsetNowTimeProvider, IDisposable
```
Comment (Aaronontheweb) [inline src/core/Akka/Actor/Scheduler/DedicatedThreadScheduler.cs]: We can just delete this class

Comment (cptjazz) [reply src/core/Akka/Actor/Scheduler/DedicatedThreadScheduler.cs]: Removed it in 7b8473ae5d3e8a84680dd2237c9880ee56cf21a7

**Comment by Aaronontheweb** [comment]
@cptjazz looks like we have some merge conflicts here - probably an issue with the large HOCON update we just merged in a few moments ago.

**Comment by cptjazz** [comment]
I'll try to rebase this PR on the latest changes.

**Pull request merged**